### ISSUE TITLE
Rate limiting and weighting recommendations 

### ISSUE THREAD

**Issue by Osmose** [mozilla/normandy#330]
https://bugzilla.mozilla.org/show_bug.cgi?id=1307004

Blocked on doing recommendations at all

**Comment by Osmose** [comment]
This is mostly a subset of the work from #329 and doesn't really need a separate issue right now.
